src/hotspot/cpu/x86/stubGenerator_x86_64_poly.cpp

```diff
@@ -1023,3 +1048,497 @@ address StubGenerator::generate_poly1305_processBlocks() {
+  // 4*5*R^4
+  __ vpsllq(YTMP1, YTMP1, 2, Assembler::AVX_256bit);
+  __ vpsllq(YTMP2, YTMP2, 2, Assembler::AVX_256bit);
+
+  //Store R^4-R for later use
+  // TODO: switch to vmovdqa
+  int _r4_save = 32*3;
+  int _r4p_save = 32*6;
+  __ vmovdqu(Address(rsp, _r4_save + 0), YMM_R0);
+  __ vmovdqu(Address(rsp, _r4_save + 32), YMM_R1);
+  __ vmovdqu(Address(rsp, _r4_save + 32*2), YMM_R2);
+  __ vmovdqu(Address(rsp, _r4p_save), YTMP1);
+  __ vmovdqu(Address(rsp, _r4p_save + 32), YTMP2);
+
+  // Get the number of multiples of 64 bytes for vectorization
+  __ movq(t0, length);
+  __ mov64(t1, 0xffffffffffffffc0);
+  __ andq(t0, t1);
+
+  // VECTOR LOOP: process 4 * 16-byte message blocks at a time
+  __ bind(L_process256Loop);
+  __ cmpl(t0, 16*4); //64 bytes (4 blocks at a time)
+  __ jcc(Assembler::belowEqual, L_process256LoopDone);
+
+  // load next block
+  poly1305_msg_mul_reduce_vec4_avx2(YMM_ACC0, YMM_ACC1, YMM_ACC2,
+                          Address(rsp, _r4_save + 0), Address(rsp, _r4_save + 32), Address(rsp, _r4_save + 32*2),
+                          Address(rsp, _r4p_save), Address(rsp, _r4p_save + 32),
+                          YTMP1, YTMP2, YTMP3, YTMP4, YTMP5, YTMP6,
+                          YTMP7, YTMP8, YTMP9, YTMP10, YTMP11, YTMP12,
+                          input, t0, t1 /*rscratch*/);
+  __ jmp(L_process256Loop);
+  // end of vector loop
+  __ bind(L_process256LoopDone);
+
+  // Need to multiply by r^4, r^3, r^2, r
+
+  //Read R^4-R; TODO: replace with vmovdqa
+  __ vmovdqu(YMM_R0, Address(rsp, _r4_r1_save + 0));
+  __ vmovdqu(YMM_R1, Address(rsp, _r4_r1_save + 32));
+  __ vmovdqu(YMM_R2, Address(rsp, _r4_r1_save + 32*2));
+
+  // YTMP1 to have bits 87-44 of all 1-4th powers of R' in 4 qwords
+  // YTMP2 to have bits 129-88 of all 1-4th powers of R' in 4 qwords
+  __ vpsllq(YTMP10, YMM_R1, 2, Assembler::AVX_256bit);
+  __ vpaddq(YTMP1, YMM_R1, YTMP10, Assembler::AVX_256bit);  //R1' (R1*5)
+  __ vpsllq(YTMP10, YMM_R2, 2, Assembler::AVX_256bit);
+  __ vpaddq(YTMP2, YMM_R2, YTMP10, Assembler::AVX_256bit);  //R2' (R2*5)
+
+  // 4*5*R
+  __ vpsllq(YTMP1, YTMP1, 2, Assembler::AVX_256bit);
+  __ vpsllq(YTMP2, YTMP2, 2, Assembler::AVX_256bit);
+
+  poly1305_mul_reduce_vec4_avx2(YMM_ACC0, YMM_ACC1, YMM_ACC2,
+                          YMM_R0, YMM_R1, YMM_R2, YTMP1, YTMP2,
+                          YTMP3, YTMP4, YTMP5, YTMP6,
+                          YTMP7, YTMP8, YTMP9, t1);
+  __ vextracti128(YTMP1, YMM_ACC0, 1);
+  __ vextracti128(YTMP2, YMM_ACC1, 1);
+  __ vextracti128(YTMP3, YMM_ACC2, 1);
+
+  __ vpaddq(YMM_ACC0, YMM_ACC0, YTMP1, Assembler::AVX_128bit);
+  __ vpaddq(YMM_ACC1, YMM_ACC1, YTMP2, Assembler::AVX_128bit);
+  __ vpaddq(YMM_ACC2, YMM_ACC2, YTMP3, Assembler::AVX_128bit);
+
+  __ vpsrldq(YTMP1, YMM_ACC0, 8, Assembler::AVX_128bit);
+  __ vpsrldq(YTMP2, YMM_ACC1, 8, Assembler::AVX_128bit);
+  __ vpsrldq(YTMP3, YMM_ACC2, 8, Assembler::AVX_128bit);
+
+  // Finish folding and clear second qword
```

Comment:
`clear second qword` doesn't apply (copy artifact from avx512); not a masked operation (still correct, just wrong comment. Subsequent `movq` will grab second qword effectively doing the same as the the avx512 code)

Reply:
done